camel/functions/search_functions.py

```diff
@@ -270,40 +332,73 @@ def summarize_text(text: str, query: str) -> str:
     return response
 
 
-def search_google_and_summarize(query: str) -> str:
-    r"""Search webs for information. Given a query, this function will use
-    the Google search engine to search for related information from the
-    internet, and then return a summarized answer.
+def continue_search(query: str, answer: str) -> bool:
+    r"""Ask LLM whether to continue search or not.
 
     Args:
         query (str): Question you want to be answered.
+        answer (str): Answer to the query.
 
     Returns:
-        str: Summarized information from webs.
+        bool: True if the user want to continue search, False otherwise.
     """
-    # Google search will return a list of urls
-    responses = search_google(query)
+    prompt = TextPrompt("Do you think the ANSWER can answer the QUERY? "
+                        "Use only 'yes' or 'no' to answer\n"
+                        "===== QUERY ====={query}\n\n"
+                        "===== ANSWER ====={answer}")
+    prompt = prompt.format(query=query, answer=answer)
+    reply = prompt_single_step_agent(prompt)
```

Comment:
What you said makes sense. However, this piece of code is used to determine whether the "answer" can adequately respond to the "query". If we were to recreate a relatively independent agent to implement this functionality, it would lead to reduced usability of the whole project. Because, as one can foresee, there are hardly any other scenarios/cases that would need to call such a specialized agent. What is more, I think the implementation here is easy to read and understand.